Pair_Up/Map.cpp

```diff
@@ -131,4 +131,245 @@ void Map::setPassScore(int score)
 std::deque<Piece> * Map::getMaplists()
 {
 	return maplists;
+}
+
+//another way to judge DeadMap
+//colorNum means a number instead a color
+bool Map::isDeadMap_()
+{
+	//from the second line(the border will be judged by another way)
+	for (int i = 1; i < 8; i++){
+		for (int j = 0; j < 9; j++)
+		{
+			//if there were any the same color Piece at the next line.
+			if (getMaplists()[i][j].colorNum == getMaplists()[i][j + 1].colorNum){
+				if ((getMaplists()[i][j].colorNum == getMaplists()[i - 1][j - 1].colorNum) || (getMaplists()[i][j].colorNum == getMaplists()[i + 1][j - 1]) || (getMaplists()[i][j].colorNum == getMaplists()[i - 1][j + 2]) || (getMaplists()[i][j].colorNum == getMaplists()[i + 1][j + 2]))
+				{
+					return false;
+				}
+				else if (j >= 2)
+				{
+					if (getMaplists()[i][j].colorNum == getMaplists()[i][j - 2])
+						return false;
+				}
+				else if (j <= 6)
+				{
+					if (getMaplists()[i][j].colorNum == getMaplists()[i][j + 2])
+						return false;
+				}
+				j++;
+			}
+			//if there were any the same color Piece between another color
+			if (j <= 6)
+			{
+				if (getMaplists()[i][j].colorNum == getMaplists()[i][j + 2].colorNum)
+				{
+					if ((getMaplists()[i][j].colorNum == getMaplists()[i - 1][j + 1].colorNum) || (getMaplists()[i][j].colorNum == getMaplists()[i + 1][j + 1].colorNum))
+						return false;
+				}
+			}
+		}
+	}
+	//from the second list to the eighth list(do not need to 
+	for (int i = 1; i < 8; i++)
+	{
+		for (int j = 0; j < 9; j++)
+		{
```

Comment:
@jushe 
It seems that it doesn't need to use two double loops to check the element, for the first loop and the second loop are nearly the same.
And you could use `maplists` instead of the `getter`.
Now you are editing a member function of `Map`

And please use break when the ***if condition*** is too long, It will make it more readable.